packages/auth/lib/AuthConnector.ts

```diff
@@ -0,0 +1,395 @@
+      throw new Error("Expired UserIdentifier.");
+    }
+
+    return userIdentifier;
+  }
+
+  private async getUserIdentifier(username: string, tenant: string): Promise<IUserIdentifier> {
+    const values = [username, tenant];
+    const { rows } = await this.authQueryHelper.adminQuery("SELECT _auth.find_user($1, $2) AS payload", values);
+
+    if (rows.length < 1 || rows[0].payload == null) {
+      throw new Error("User not found!");
+    }
+
+    return rows[0].payload;
+  }
+
+  // tslint:disable-next-line:prettier
+  public async createUserAuthentication(dbClient: PgClient, userId: string, isActive: boolean, loginProviderSets: string[], modifyProviderSets: string[], authFactorCreationTokens: string[]): Promise<ILoginData> {
+    try {
+      const authFactorCreations: IAuthFactorCreation[] = authFactorCreationTokens.map(this.decryptAuthFactorCreationToken.bind(this));
+
+      await this.authQueryHelper.setAdmin(dbClient);
+
+      const values = [userId, isActive, loginProviderSets, modifyProviderSets, JSON.stringify(authFactorCreations)];
+      const { rows } = await dbClient.query(`SELECT _auth.create_user_authentication($1, $2, $3, $4, $5) AS payload;`, values);
+
+      await this.authQueryHelper.unsetAdmin(dbClient);
+
+      if (rows.length < 1 || rows[0].payload == null) {
+        throw new Error("Incorrect response from create_user_authentication.");
+      }
+
+      const result = rows[0].payload;
+      const issuedAtLuxon = DateTime.fromMillis(result.issuedAt, { zone: "UTC" });
+
+      const loginData: ILoginData = {
+        accessToken: this.cryptoFactory.encrypt(result.accessToken),
+        refreshToken: result.refreshToken != null ? this.cryptoFactory.encrypt(result.refreshToken) : null,
+        tokenMeta: {
+          userId: result.userId,
+          providerSet: result.providerSet,
+          issuedAt: issuedAtLuxon.toISO(),
+          expiresAt: issuedAtLuxon.plus({ seconds: result.accessTokenMaxAgeInSeconds }).toISO(),
+          accessTokenMaxAgeInSeconds: result.accessTokenMaxAgeInSeconds
+        }
+      };
+
+      return loginData;
+    } catch (err) {
+      this.logger.trace("createUserAuthentication.failed", err);
+      throw new Error("Create UserAuthentication failed.");
+    }
+  }
+
+  public createAuthFactorCreationToken(authFactorCreation: IAuthFactorCreation): string {
+    return this.encryptAuthFactorCreationToken(authFactorCreation);
+  }
+
+  public createAuthFactorProofToken(authFactorProof: IAuthFactorProof): string {
+    return this.encryptAuthFactorProofToken(authFactorProof);
+  }
+
+  public async findUser(username: string, tenant: string): Promise<IFindUserResponse> {
+    let userIdentifier: IUserIdentifier = null;
+    let isFake = false;
+
+    try {
+      userIdentifier = await this.getUserIdentifier(username, tenant);
+    } catch (err) {
+      /* Ignore Error */
+    }
+
+    // This function will always return some encrypted IDs
```

Comment:
The function/method `findUser` will always return some IDs. And thats what the lines after this comment are doing. If anything is null it will generate fake IDs.

Reply:
Jupp, totally correct. Still those lines of code are not a function as stated in the comment, but only a bunch of lines. However, it is not critical i suppose.